manager/api/transport_test.go

```diff
@@ -0,0 +1,535 @@
+		channels := []manager.Channel{}
+		for i := 0; i < 10; i++ {
+			channel := validChannel
+			chanID, _ := svc.CreateChannel(validUser.Email, channel)
+			channel.ID = chanID
+			channels = append(channels, channel)
+		}
+
+		cases := []struct {
+			desc   string
+			auth   string
+			status int
+			res    []manager.Channel
+		}{
+			{"get a list of channels", validUser.Email, http.StatusOK, channels},
+			{"try to get a list of channels with invalid token", invalidUser.Email, http.StatusForbidden, nil},
+		}
+
+		for _, tc := range cases {
+			res, err := sendRequest(client, http.MethodGet, fmt.Sprintf("%s/channels", url), "", tc.auth, nil)
+			assert.Nil(t, err, fmt.Sprintf("%s: unexpected error %s", tc.desc, err))
+			var body map[string][]manager.Channel
+			json.NewDecoder(res.Body).Decode(&body)
+			assert.Equal(t, tc.status, res.StatusCode, fmt.Sprintf("%s: expected status code %d got %d", tc.desc, tc.status, res.StatusCode))
+			assert.ElementsMatch(t, tc.res, body["channels"], fmt.Sprintf("%s: expected body %s got %s", tc.desc, tc.res, body["channels"]))
+		}
+	})
+}
+
+func TestRemoveChannel(t *testing.T) {
+	runService(func(svc manager.Service, url string, client *http.Client) {
+		svc.Register(validUser)
+		chanID, _ := svc.CreateChannel(validUser.Email, validChannel)
+		validChannel.ID = chanID
+
+		cases := []struct {
+			desc   string
+			chanID string
+			auth   string
+			status int
+		}{
+			{"remove existing channel", validChannel.ID, validUser.Email, http.StatusNoContent},
+			{"try to remove non-existent channel", validChannel.ID, validUser.Email, http.StatusNoContent},
+			{"try to remove channel with invalid id", "123e4567-e89b-12d3-a456-000000000042", validUser.Email, http.StatusNoContent},
+			{"try to remove channel with invalid token", validChannel.ID, invalidUser.Email, http.StatusForbidden},
+		}
+
+		for _, tc := range cases {
+			res, err := sendRequest(client, http.MethodDelete, fmt.Sprintf("%s/channels/%s", url, tc.chanID), "", tc.auth, nil)
+			assert.Nil(t, err, fmt.Sprintf("%s: unexpected error %s", tc.desc, err))
+			assert.Equal(t, tc.status, res.StatusCode, fmt.Sprintf("%s: expected status code %d got %d", tc.desc, tc.status, res.StatusCode))
+		}
+	})
+}
+
+func TestConnect(t *testing.T) {
+	runService(func(svc manager.Service, url string, client *http.Client) {
+		svc.Register(validUser)
+		clientID, _ := svc.AddClient(validUser.Email, validClient)
+		chanID, _ := svc.CreateChannel(validUser.Email, validChannel)
+
+		cases := []struct {
+			desc     string
+			chanID   string
+			clientID string
+			auth     string
+			status   int
+		}{
+			{"connect existing client to existing channel", chanID, clientID, validUser.Email, http.StatusOK},
+			{"try to connect existing client to non-existent channel", "123e4567-e89b-12d3-a456-000000000042", clientID, validUser.Email, http.StatusNotFound},
+			{"try to connect client with invalid id to channel", chanID, "1", validUser.Email, http.StatusNotFound},
```

Comment:
Add tests that will try to connect client from user A to the channel from user B and vice versa.